### DISCUSSION TITLE
Swagger ui not working when quarkus.vertx.classpath-resolving=false

### DISCUSSION THREAD

**Discussion by nawaz081** [quarkusio/quarkus#41850] [Community]
I have a problem similar to the one mentioned in the below stackoverflow thread - did anyone find a workaround?

https://stackoverflow.com/questions/78709361/swagger-ui-not-working-in-prod-mode-when-quarkus-vertx-classpath-resolving-is-di

**Comment by phillip-kruger** [comment]
There is a workaround mentioned in the link above:

"there was a workaround in the end, and it was as you suggested. I copied all the swagger-ui html files in my resources/META-INF/resources folder and only had to adjust one file (swagger-initializer.js) by adjusting the SwaggerUIBundle url property"

**Reply by nawaz081** [reply]
I don't have any of those swagger specific files other than the quarkus swagger ui lib in my pom

**Reply by phillip-kruger** [reply]
They will be in the final artifact after you build your project

**Reply by nawaz081** [reply]
In my local workspace I couldn't find and also we deploy to AWS containers and I couldn't find the files there as well. I am unable to post in stackoverflow to ask Emile regarding the same. Is there a sample project I can take a look at to see the files?

**Reply by phillip-kruger** [reply]
You can have a look at: https://github.com/quarkusio/todo-demo-app

There, we include swagger ui with property `quarkus.swagger-ui.always-include=true`. When building with `mvn clean install` you can find the swagger files in `/target/quarkus-app/quarkus/generated-bytecode.jar` in `META-INF/io.smallrye_smallrye-open-api-ui__jar/META-INF/resources/openapi-ui/`

If you copy the files you want to change to your own resource folder if will load that file first.

**Reply by nawaz081** [reply]
@phillip-kruger - Thanks so much for pointing me to the correct location of swagger files. Emile has mentioned in his solution that he changed swagger-initializer.js but I don't have that file in my target folder instead I have swagger-ui-bundle.js but don't see the property he mentioned in his solution.

I have copied the html files and the css/js files required to my resources/META-INF/resources but still getting the same error, probably because I have not set the url-property correctly.

**Reply by phillip-kruger** [reply]
What error are you getting ? The problem in the link above is that the files are not included because vertx scanning is disabled. Maybe explain your issue as it seems different from the stackoverflow one.

**Reply by nawaz081** [reply]
Everything is same interms of the issue mentioned in the stack overflow thread except the thing that in the solution Emile mentioned he has modified the swagger-initializer.js and I am unable to find that file in my swagger files (basically at the location where you shared in target folder)

**Reply by phillip-kruger** [reply]
So you are also setting `quarkus.vertx.classpath-resolving=false` ?

**Reply by nawaz081** [reply]
yes

**Reply by nawaz081** [reply]
quarkus.vertx.caching=false
quarkus.vertx.classpath-resolving=false

with this config if I try accessing url https://devurl-sample/openapi - it throws 404 not found error but without disabling quarkus.vertx.classpath-resolving, it works fine.

**Reply by phillip-kruger** [reply]
Ok, so the issue is not that swagger ui does not work. swagger-ui probably loads, but also fails as it can not find the openapi schema ? You can also include the openapi schema document as a static document in your final artifact. (However, you will not able to use OASFilters that runs at runtime)  You can save a copy of the generated schema document to your project using this: https://quarkus.io/guides/openapi-swaggerui#quarkus-smallrye-openapi_quarkus-smallrye-openapi-store-schema-directory

**Reply by nawaz081** [reply]
I have tried the above solution as well and still getting the same error 404 when accessing https://devurl-sample/openapi

**Reply by phillip-kruger** [reply]
Best way to move forward is to create and share a reproducer. Thanks :)
